draft-bbs-signatures.md

```diff
@@ -738,6 +748,76 @@ Procedure:
 5. return (scalar_1, ..., scalar_n)
 ```
 
+### OctetsToSignature
+
+This operation describes how to decode an octet string, validate it and return the underlying components that make up the signature.
+
+```
+(A, e, s) = octets_to_signature(signature_octets)
+
+Inputs:
+
+- signature_octets, octet string.
+
+Parameters:
+
+- octet_point_length, the length in bytes of the octet string output by the point_to_octets function
+
+Outputs:
+
+- A map comprised of the underlying signature components `A` `e` and `s`, OR the result INVALID
+
+Procedure:
+
+1. if len(signature_octets) != (octet_point_length + 2 * xof_no_of_bytes), return INVALID
+
+2. a_octets = signature_octets[0..(octet_point_length - 1)]
+
+3. A = octets_to_point(a_octets)
+
+4. if A is INVALID, return INVALID // TODO Check A in other ways here should we be calling IsValidPoint?
+
+5. index = octet_point_length
+
+5. e = OS2IP(signature_octets[index..(index + xof_no_of_bytes - 1])) mod q
+
+6. index += xof_no_of_bytes
+
+6. s = OS2IP(signature_octets[index..(index + xof_no_of_bytes - 1)]) mod q
+
+6. return (A, e, s)
+```
+
+### SignatureToOctets
+
+This operation describes how to encode a signature to an octet string.
+
+```
+signature_octets = signature_to_octets(A, e, s)
+
+Inputs:
```

Comment:
If we assume A e and s are well formed then we are implicitly assuming well formed inputs vs untrusted inputs on the OctetsToSignature operation

Reply:
Add a note that documents this is the case because only invocation of this operation is via the Sign method which already checks A e and s.